numpy/f2py/cfuncs.py

```diff
@@ -684,126 +686,160 @@
     Py_XDECREF(tmp);
     {
         PyObject* err = PyErr_Occurred();
-        if (err==NULL) err = #modulename#_error;
-        PyErr_SetString(err,errmess);
+        if (err == NULL) {
+            err = #modulename#_error;
+        }
+        PyErr_SetString(err, errmess);
```

Comment:
Is this legal? Shouldn't this be
```suggestion
        PyErr_SetString(Py_TYPE(err), errmess);
```
or similar?

Reply:
I wondered about that too, but it turns out that `#modulename#_error` is a variable pointing to an exception object. It is set in `numpy/f2py/rules.py`.
```
\t#modulename#_error = PyErr_NewException (\"#modulename#.error\", NULL, NULL);
```
That's another file that could use a cleanup. The intent of this PR was just to replace some compatibility macros, the reformatting was so that I could read it :) It could use more...

Reply:
Oh, I guess my suggestion was wrong too. I meant for the case when `err != NULL`, and holds the currently active exception.

Reply:
`PyErr_Occurred` returns the error type.

>Test whether the error indicator is set. If set, return the exception type

Reply:
I think what is wanted here is a chained error, but that is for another PR.